File: service/src/main/java/com/coopra/service/service_implementations/ActivitiesService.java
package com.coopra.service.service_implementations;

import com.coopra.data.DashboardActivityEnvelope;
import com.coopra.service.RetrofitHelper;
import com.coopra.service.interfaces.ActivitiesInterface;

import retrofit2.Call;
import retrofit2.Callback;

public class ActivitiesService {
    public static void getFeedTracks(String token, Callback<DashboardActivityEnvelope> callback) {
        ActivitiesInterface service = RetrofitHelper.createRetrofit().create(ActivitiesInterface.class);

        Call<DashboardActivityEnvelope> call = service.getFeedTracks(token);
        call.enqueue(callback);
    }

    public static void getNextTracks(String token, String url, Callback<DashboardActivityEnvelope> callback) {
        ActivitiesInterface service = RetrofitHelper.createRetrofit().create(ActivitiesInterface.class);

        Call<DashboardActivityEnvelope> call = service.getNextTracks(url, token);
        call.enqueue(callback);
    }
}

File: service/src/main/java/com/coopra/service/service_implementations/ActivitiesService.kt
package com.coopra.service.service_implementations

import com.coopra.data.DashboardActivityEnvelope
import com.coopra.service.RetrofitHelper
import com.coopra.service.interfaces.ActivitiesInterface

class ActivitiesService {
    private val retrofitHelper = RetrofitHelper()

    suspend fun getFeedTracks(token: String): DashboardActivityEnvelope {
        val service = retrofitHelper.createRetrofit().create(ActivitiesInterface::class.java)
        return service.getFeedTracks(token)
    }

    suspend fun getNextTracks(token: String, url: String): DashboardActivityEnvelope {
        val service = retrofitHelper.createRetrofit().create(ActivitiesInterface::class.java)
        return service.getNextTracks(url, token)
    }
}